src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.CurlResponseMessage.cs

```diff
@@ -0,0 +1,500 @@
+                _easy = easy;
+
+                _responseStream = new CurlResponseStream(this);
+                RequestMessage = easy.RequestMessage;
+                Content = new StreamContent(_responseStream);
+            }
+
+            internal CurlResponseStream ResponseStream
+            {
+                get { return _responseStream; }
+            }
+        }
+
+        /// <summary>
+        /// Provides a response stream that allows libcurl to transfer data asynchronously to a reader.
+        /// When writing data to the response stream, either all or none of the data will be transferred,
+        /// and if none, libcurl will pause the connection until a reader is waiting to consume the data.
+        /// Readers consume the data via ReadAsync, which registers a read state with the stream, to be
+        /// filled in by a pending write.  Read is just a thin wrapper around ReadAsync, since the underlying
+        /// mechanism must be asynchronous to prevent blocking libcurl's processing.
+        /// </summary>
+        private sealed class CurlResponseStream : Stream
+        {
+            /// <summary>A cached task storing the Int32 value 0.</summary>
+            private static readonly Task<int> s_zeroTask = Task.FromResult(0);
+
+            /// <summary>
+            /// A sentinel object used in the <see cref="_completed"/> field to indicate that
+            /// the stream completed successfully.
+            /// </summary>
+            private static readonly Exception s_completionSentinel = new Exception("s_completionSentinel");
+
+            /// <summary>A object used to synchronize all access to state on this response stream.</summary>
+            private readonly object _lockObject = new object();
+
+            /// <summary>The response message that owns this response stream.</summary>
+            private readonly CurlResponseMessage _parentResponse;
+
+            /// <summary>Stores whether Dispose has been called on the stream.</summary>
+            private bool _disposed = false;
+
+            /// <summary>
+            /// Null if the Stream has not been completed, non-null if it has been completed.
+            /// If non-null, it'll either be the <see cref="s_completionSentinel"/> object, meaning the stream completed
+            /// successfully, or it'll be an Exception object representing the error that caused completion.
+            /// That error will be transferred to any subsequent read requests.
+            /// </summary>
+            private Exception _completed;
+
+            /// <summary>
+            /// The state associated with a pending read request. When a reader requests data, it puts
+            /// its state here for the writer to fill in when data is available.
+            /// </summary>
+            private ReadState _pendingReadRequest;
+
+            /// <summary>How much data has been read.</summary>
+            private long _position = 0;
```

Comment:
I had it hear because it was useful in debugging (note that it was both being incremented and traced out in multiple VerboseTrace calls).  I was also hoping we could return this from the stream's get_Position.  But since we don't do that on Windows, and since it was only for debugging purposes, I removed the field.  We could choose to add it back later, potentialy under a #if CURLHANDLER_VERBOSE.